hawkbit-repository/hawkbit-repository-test/src/main/java/org/eclipse/hawkbit/repository/test/util/PostgreSqlTestDatabase.java

```diff
@@ -8,61 +8,55 @@
+    protected static final String POSTGRESQL_URI_PATTERN = "jdbc:postgresql://{host}:{port}/{path}?currentSchema={db}*";
 
     @Override
-    protected void createSchemaUri() {
-        schemaName = "sp" + RandomStringUtils.randomAlphanumeric(10).toLowerCase();
-        this.uri = this.uri.substring(0, uri.indexOf('?'));
-
-        System.setProperty("spring.datasource.url", uri + "?currentSchema=" + schemaName);
+    protected boolean isApplicable() {
+        return "POSTGRESQL".equals(System.getProperty("spring.jpa.database")) //
+                && MATCHER.match(POSTGRESQL_URI_PATTERN, System.getProperty(SPRING_DATASOURCE_URL_KEY))  //
+                && !StringUtils.isEmpty(getSchemaName());
     }
 
     @Override
-    protected boolean isRunningWithSql() {
-        return "POSTGRESQL".equals(System.getProperty("spring.jpa.database"));
+    protected void createSchema() {
+        final String schemaName = getSchemaName();
+        final String uri = System.getProperty(SPRING_DATASOURCE_URL_KEY);
+        LOGGER.info("\033[0;33m Creating postgreSql schema {} if not existing \033[0m", schemaName);
+
+        executeStatement(uri.split("\\?currentSchema=")[0], "CREATE SCHEMA IF NOT EXISTS " + schemaName + ";");
     }
 
     @Override
-    protected void createSchema() {
-        try (Connection connection = DriverManager.getConnection(uri, username, password)) {
-            try (PreparedStatement statement = connection.prepareStatement("CREATE schema " + schemaName + ";")) {
-                LOG.info("Creating schema {} on uri {}", schemaName, uri);
-                statement.execute();
-                LOG.info("Created schema {} on uri {}", schemaName, uri);
-            }
-        } catch (final SQLException e) {
-            LOG.error("Schema creation failed!", e);
-        }
+    protected void dropSchema() {
+        final String schemaName = getSchemaName();
+        final String uri = System.getProperty(SPRING_DATASOURCE_URL_KEY);
 
+        executeStatement(uri.split("\\?currentSchema=")[0], "DROP schema " + schemaName + " CASCADE;");
     }
 
     @Override
-    protected void dropSchema() {
-        try (Connection connection = DriverManager.getConnection(uri, username, password)) {
-            try (PreparedStatement statement = connection.prepareStatement("DROP schema " + schemaName + " CASCADE;")) {
-                LOG.info("Dropping schema {} on uri {}", schemaName, uri);
-                statement.execute();
-                LOG.info("Dropped schema {} on uri {}", schemaName, uri);
-            }
-        } catch (final SQLException e) {
-            LOG.error("Schema drop failed!", e);
-        }
+    protected String getRandomSchemaUri() {
+        final String schemaName = "HAWKBIT_TEST" + RandomStringUtils.randomAlphanumeric(10).toLowerCase();
+        final String uri = System.getProperty(SPRING_DATASOURCE_URL_KEY);
+
+        return uri.substring(0, uri.indexOf('?')) + "?currentSchema=" + schemaName;
```

Comment:
With our default values, the initial run has a URL without `currentSchema` configured. But also every run with the additional parameter will work:
`        if (uri.contains("?")) {
            return uri.substring(0, uri.indexOf('?')) + "?currentSchema=" + schemaName;
        } else {
            return uri + "?currentSchema=" + schemaName;
        }`